src/banman.cpp

```diff
@@ -45,16 +45,17 @@ void BanMan::DumpBanlist()
 
     banmap_t banmap;
     {
-        LOCK(m_cs_banned);
+        LOCK(m_banned_mutex);
         SweepBanned();
-        if (!BannedSetIsDirty()) return;
+        if (!m_is_dirty) return;
         banmap = m_banned;
-        SetBannedSetDirty(false);
+        m_is_dirty = false;
     }
 
     int64_t n_start = GetTimeMillis();
     if (!m_ban_db.Write(banmap)) {
-        SetBannedSetDirty(true);
+        LOCK(m_banned_mutex);
+        m_is_dirty = true;
```

Comment:
**Doubt:** Why did you prefer this way of writing code over using `WITH_LOCK`, which you discussed with @MarcoFalke in the earlier comment.

```diff
--- a/src/banman.cpp
+++ b/src/banman.cpp
@@ -45,11 +45,12 @@ void BanMan::DumpBanlist()
         SweepBanned();
         if (!m_is_dirty) return;
         banmap = m_banned;
+        m_is_dirty = false;
     }
 
     int64_t n_start = GetTimeMillis();
-    if (m_ban_db.Write(banmap)) {
-        WITH_LOCK(m_banned_mutex, m_is_dirty = false);
+    if (!m_ban_db.Write(banmap)) {
+        WITH_LOCK(m_banned_mutex, m_is_dirty = true);
     }
 
     LogPrint(BCLog::NET, "Flushed %d banned node addresses/subnets to disk  %dms\n", banmap.size(),
```

Reply:
With `m_is_dirty = false;` a few lines above I think it's more readable, no?

Reply:
Makes sense. So it was an aesthetic choice, which increases code readability, and both of these don't have any functional difference. Am I correct in my analysis?

Reply:
Yes, you are.

Reply:
Got it. Thanks, @hebasto!